### PR TITLE
Get distance Between Actor and a Scene Component

### PR DESCRIPTION
**Pull request by FXIce18**
Added a Scene Component as Input so we can get the Distance between Actor1 and the SceneComponent if it is not null, instead of the distance between Actor1 and Actor2.

SceneComponent property is set in "Input" Category so the State Tree can compile even if the SceneComponent is not set.

### CODE REVIEW

File: Source/AIExtensions/Public/StateTree/Tasks/AIExtStateTreeTaskGetDistanceBetweenActors.h [hunk 1]
```diff
@@ -22,6 +22,10 @@ struct AIEXTENSIONS_API FAIExtStateTreeTaskGetDistanceBetweenActorsInstanceData
     UPROPERTY( EditAnywhere, Category = "Context" )
     TObjectPtr< AActor > Actor2 = nullptr;
 
+    /** The Scene Component */
```
Comment (TheEmidee) [inline Source/AIExtensions/Public/StateTree/Tasks/AIExtStateTreeTaskGetDistanceBetweenActors.h]: add in the comment that the component has precedence over Actor2

File: Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskGetDistanceBetweenActors.cpp [hunk 2]
```diff
@@ -10,7 +10,11 @@ EStateTreeRunStatus FAIExtStateTreeTaskGetDistanceBetweenActors::Tick( FStateTre
 
     auto & instance_data = context.GetInstanceData( *this );
 
-    if ( instance_data.Actor1 != nullptr && instance_data.Actor2 != nullptr )
+    if ( instance_data.Actor1 != nullptr && instance_data.OtherComponent != nullptr )
```
Comment (TheEmidee) [inline Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskGetDistanceBetweenActors.cpp]: I would check 

```
if ( instance_data.Actor1 == nullptr )
{
    return EStateTreeRunStatus::Failed;
}

if ( instance_data.OtherComponent != nullptr )
{
    // ...
}
else if ( instance_data.Actor2 != nullptr )
{
    // ...
}
else
{
    return EStateTreeRunStatus::Failed;
}
```

**Review by TheEmidee — CHANGES_REQUESTED** [review]

File: Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskGetDistanceBetweenActors.cpp (hunk 2 above)
Comment (FXIce18) [reply Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskGetDistanceBetweenActors.cpp]: Can Context parameters be nullptr ?
State Tree doesn't compile if parameter is not Bound.

Comment (TheEmidee) [reply Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskGetDistanceBetweenActors.cpp]: even if their own tasks in the engine they check for that so I guess it's a good habit :)

**Pull request merged**